lib/bas/bot/format_do_bill_alert.rb

```diff
@@ -48,17 +48,17 @@ class FormatDoBillAlert < Bot::Base
     # read function to execute the PostgresDB Read component
     #
     def read
-      reader = Read::Postgres.new(read_options.merge(conditions))
+      reader = Read::Postgres.new(read_options.merge(combined_conditions))
 
-      reader.execute
+      @previous_billing_data, @current_billing_data = split_billing_data(reader.execute)
```

Comment:
The main idea of the read step is to get only one record from the database. Since this is the format bot, we want the last record fetched from the digital ocean (the "unprocessed" record). What do you think if we update the `read` function of the `fetch_billing_from_digital_ocean` bot, read the last `processed` records, and add it to the bot response? In this way, the write step of the `fetch_billing_from_digital_ocean` bot could add a record of the form `{ success: { billing: { ... }, last_billing: { ... } } }` where `last_billing` will have the processed record. Then, in the `format_do_bill_alert` bot, you will have access to that value. There is a similar process for the `fetch_birthday_from_notion` bot. Let me know what you think.